assets/js/app/Panel.js

```diff
@@ -798,6 +798,44 @@ LocusZoom.Panel.prototype.render = function(){
 };
 
 
+// Generate a chart-consumable ticks array from an arbitrary custom layout ticks definition.
+// Will pass through arrays but parse objects that invoke scale functions to generate ticks
+// from a data set (e.g. generating chromosome ticks for a GWAS manhattan plot).
+LocusZoom.Panel.prototype.generateCustomTicks = function(axis_layout){
```

Comment:
I haven't tried this with the pheWAS data layer just yet, but in principle I like this method better than my rough PheWAS approach, as it offers an option for the panel to customize tick display with defaults.  As far as I can tell, axis configuration is mostly the responsibility of the panel.

For PheWAS, our use case encompassed categories for colors as well as ticks. My current implementation relies on a new data_layer `category_field` option; for performance reasons I think it would be good to have the category data in a format that supported both use cases. (that way we don't need to iterate over all data so many times)  

I'd be happy to refactor my code as needed.